File: backend/Player.cs
using Newtonsoft.Json;
using System;
using System.IO;
using System.Linq;
using System.Net;

namespace backend
{
    public class Player
    {
        public string Name { get; set; }
        public string SteamName { get; set; }
        public bool ProfilePublic { get; set; }
        public string SteamID { get; set; }
        public SkillLevel Skill { get; set; }
    }
}

File: frontend/lib/model/player.dart
import 'package:frontend/model/skill.dart';

class Player {
  String name;
  String steamName;
  String steamID;
  Skill skill;

  bool isSelected;

  Player(
      {this.name,
      this.steamName = "Player1",
      this.steamID = "",
      this.isSelected = false,
      this.skill}) {
    if (skill == null) {
      skill = Skill();
    }
  }

  Player.fromJson(dynamic json) {
    name = json["name"] != null ? json["name"] : "no name found";
    steamName = json["steamName"] != null ? json["steamName"] : "Player1";
    steamID = json["steamID"] != null ? json["steamID"] : ["invalid id"];
    skill = Skill.fromJson(json["skill"]);
  }

  Map<String, dynamic> toJson() => {
        'name': name,
        'steamName': steamName,
        'steamID': steamID,
        'skill': skill.toJson(),
      };
}
